### PR TITLE
Re-organization of code for handling Nodes - Simplification and improving tests

### PR DESCRIPTION
**Pull request by olivier-stasse**
This PR is a refactoring of the code when handling Nodes.
The main modification are simplification and making SotLoaderBasic and SotLoader classes inheriting from Nodes.
Tests are improved and should be passing.

### CODE REVIEW

**Review by nim65s — COMMENTED** [review]
Please try to give some time for reviews before merging.

You are adding a dependency here, which is a breaking change with huge implications on the packaging. At first glance, I don't understand why you did that, could you document it a bit ? Also, I think you forgot to update package.xml with this change.

File: src/sot_loader_basic.cpp [hunk 1]
```diff
@@ -13,9 +13,14 @@
...
 
+#include <boost/make_shared.hpp>
+#include <boost/shared_ptr.hpp>
 #include <exception>
+#include <sot/core/robot-utils.hh>
+#include <stdexcept>
 
-#include "dynamic_graph_bridge/ros_parameter.hpp"
+#include "pinocchio/multibody/model.hpp"
+#include "pinocchio/parsers/urdf.hpp"
```
Comment (nim65s) [inline src/sot_loader_basic.cpp:23]: What, pinocchio ? This is another huge change here. And you also need to add this in package.xml & CMakeLists.txt

**Comment by nim65s** [comment]
Also, I might be wrong, but I don't see any `target_link_libraries` to either urdfdom or pinocchio, which is another error here

File: src/sot_loader_basic.cpp (hunk 1 above)
Comment (olivier-stasse) [reply src/sot_loader_basic.cpp:23]: Bumping my head on this, there is an indirect dependency on pinocchio through the use of ```robot-utils```.
The problem was related to boost::variant and I found a way to fix it without including again pinocchio.

**Pull request merged**